frontend/lib/resolution/default-functions.cpp

```diff
@@ -855,9 +862,15 @@ generateCastToEnum(Context* context,
   If no method was generated, returns nullptr.
 */
 const TypedFnSignature*
-getCompilerGeneratedMethod(Context* context, const Type* type,
+getCompilerGeneratedMethod(Context* context, const QualifiedType receiverType,
                            UniqueString name, bool parenless) {
-  return getCompilerGeneratedMethodQuery(context, type, name, parenless);
+  // Normalize recieverType to have either the TYPE, PARAM, or VAR Kind.
+  // This should allow for better re-use of the Query.
+  auto qt = receiverType;
+  if (!qt.isType() && qt.isParam()) {
+    qt = QualifiedType(QualifiedType::VAR, qt.type(), qt.param());
```

Comment:
if it's not a type but it is a param, it still has the param value, but it is irrelevant / unused since it's not QualifiedType::PARAM

Also I was expecting you'd use VAR for everything unless it's a c_ptr type here, but at least ruling out many query runs for different params seems worthwhile